### DISCUSSION TITLE
Multi Page Apps Using New API

### DISCUSSION THREAD

**Discussion by unnamedunknownusername** [GenieFramework/Genie.jl#602] [Q&A] [answered]
I love the new simplistic API. however I was wondering about multi-page apps in the new API/GenieBuilderVer0.2.

say I have just created a new app with GenieBuilderVer0.2

so I have `app.jl`
```.jl
module App

using GenieFramework
@genietools

@handlers begin
  @out message = "Hello There"

  @onchange isready begin
    @show "App is loaded"
  end
end

@page("/", "app.jl.html")

end

```

and I have `app.jl.html`
```html
<h1>{{message}}</h1>
<p>This is the only default view for the application.</p>
<p>You can change this view by editing the file <code>app.jl.html</code>.</p>



```

and I add a `page1.jl` file that looks like this
```.jl
module   Page1

using GenieFramework
@genietools

@handlers begin
  @out message = "Hello Page1"

  @onchange isready begin
    @show "Page1 is loaded"
  end
end

@page("/page1", "page1.jl.html")

end
```
and I add a `page1.jl.html` which looks like this
```html
<h1>{{message}}</h1>
<p>This is the only default view for the application.</p>
<p>You can change this view by editing the file <code>page1.jl.html</code>.</p>

```
so now in `MyApp` folder I will have 
 - `app.jl`
 - `app.jl.html`
 - `page1.jl`
 - `page1.jl.html`

I was thinking that the `@page("/page1", "page1.jl.html")` line would add the new page and route to the `Pages[]` list and thus be accessible when I go to its associated URL.
but when I try and reload/restart the app and go to the `/page1`  URL it doesn't seem to work.
![image](https://user-images.githubusercontent.com/109242984/204101945-55f0c1b6-b7f6-4cd7-ac3b-1147d42c6adf.png)
and the new URL does not show up in the side bar where I see the `/` URL

So how would I go about creating a multi-page app using the new minimal API.

Sorry if it is a beginner question, just wanted to see if something like this was possible.

**Comment by essenciary** [comment, marked as answer]
It's a good question. The obvious issue is that the application does not know about your `page1.jl` file. The app automatically loads the `app.jl` file as the entry point -- but beyond this point, you need to load any other file yourself. So add `include("page1.jl") inside app.jl.

**Reply by unnamedunknownusername** [reply]
Thanks for the prompt reply, and thanks so much for all the hard work on the Genie ecosystem. That makes a lot of sense, can't believe I missed it.

**Reply by essenciary** [reply]
Let me know if that works.

**Reply by unnamedunknownusername** [reply]
Sorry for the delay, I forgot to post this after getting it to work
![Screenshot_20221216_054218](https://user-images.githubusercontent.com/109242984/208265266-ab3b7335-9a83-432e-82c1-1d2857056fcc.png)

yes "including" the files works and both the URL's show up in the side bar where I see now see the `/` URL as well as the `/page1` URL

**Reply by AbhimanyuAryan** [reply]
thanks for posting the solution. If the project is OSS. Please post the link to git repo as well. I am thinking maintaining a repository of GB projects on GIt and then we can showcase them on website 👍🏻
